compose/snippets/src/main/java/com/example/compose/snippets/animations/sharedelement/CustomizeSharedElementsSnippets.kt

```diff
@@ -628,3 +636,91 @@ fun PlaceholderSizeAnimated_Demo() {
     }
 // [END android_compose_shared_element_placeholder_size]
 }
+
+private sealed class Screen {
+    data object Home : Screen()
+    data class Details(val id: Int) : Screen()
+}
+
+@Preview
+@Composable
+fun CustomPredictiveBackHandle() {
+    // [START android_compose_shared_element_custom_seeking]
+    val seekableTransitionState = remember {
+        SeekableTransitionState<Screen>(Screen.Home)
+    }
+    val transition = rememberTransition(transitionState = seekableTransitionState)
+
+    PredictiveBackHandler(seekableTransitionState.currentState is Screen.Details) { progress ->
+        try {
+            progress.collect { backEvent ->
+                // code for progress
+                try {
+                    seekableTransitionState.seekTo(backEvent.progress, targetState = Screen.Home)
+                } catch (e: CancellationException) {
+                    // ignore the cancellation
+                }
+            }
+            // code for completion
+            seekableTransitionState.animateTo(seekableTransitionState.targetState)
+        } catch (e: CancellationException) {
+            // code for cancellation
+            seekableTransitionState.animateTo(seekableTransitionState.currentState)
+        }
```

Comment:
I find it slightly confusing why do I need to suppress the cancellation exceptions here.
Can we explain it a bit more in the comments? 🙏🏼

Reply:
Added comments.